gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -448,15 +465,18 @@ private void sendChallenge(FactorProvider factorProvider, RoutingContext routing
                             error -> handler.handle(Future.failedFuture(error))
                     );
         } else {
-            sendChallenge(factorProvider, factorContext, handler);
+            sendChallenge(factorProvider, factorContext, endUser, client, factor, handler);
         }
     }
 
-    private void sendChallenge(FactorProvider factorProvider, FactorContext factorContext, Handler<AsyncResult<Void>> handler){
+    private void sendChallenge(FactorProvider factorProvider, FactorContext factorContext, User endUser, Client client, Factor factor, Handler<AsyncResult<Void>> handler) {
         factorProvider.sendChallenge(factorContext)
                 .subscribeOn(Schedulers.io())
                 .subscribe(
-                        () -> handler.handle(Future.succeededFuture()),
+                        () -> {
+                            updateAuditLog(MFA_CHALLENGE_SEND, endUser, client, factor, factorContext, null);
+                            handler.handle(Future.succeededFuture());
+                        },
                         error -> handler.handle(Future.failedFuture(error))
```

Comment:
missing log?

Reply:
Error log is present in line around 229. This sendChallenge method is private and overridden. Eventually there is a block of code which handle once the feature is failed.
`if (resChallenge.failed() && error == null) {
                    logger.error("An error has occurred when sending MFA challenge", resChallenge.cause());
                    routingContext.fail(resChallenge.cause());
                    return;
                }`

Reply:
sorry for the shortcut I thought about the Audit log. It is not possible to get the Audit log MFA_CHALLENGE_SENT with the status Failed. Does it make sense to add it here?